src/app/block/[block]/page.tsx

```diff
@@ -1,5 +1,105 @@
-const BlockPage = ({ params: { block } }: { params: { block: string } }) => (
-  <h1>{block}</h1>
-);
+import { createPublicClient, http, formatEther } from 'viem';
+import { optimism } from 'viem/chains';
+import Link from 'next/link';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
 
-export default BlockPage;
+const client = createPublicClient({
+  chain: optimism,
+  transport: http(process.env.RPC_URL),
+});
+
+function formatTimestamp(timestamp: number): string {
+  const date = new Date(timestamp * 1000);
+  const now = new Date();
+  const diffTime = Math.abs(now.getTime() - date.getTime());
+  const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+
+  const formattedDate = date.toLocaleString('en-US', {
+    month: 'short',
+    day: '2-digit',
+    year: 'numeric',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    hour12: true,
+    timeZone: 'UTC'
+  });
+
+  return `${diffDays} day${diffDays !== 1 ? 's' : ''} ago (${formattedDate} +UTC)`;
+}
+
+async function BlockPage({ params: { block } }: { params: { block: string } }) {
+  const blockNumber = BigInt(block.toString());
+  
+  const blockData = await client.getBlock({
+    blockNumber: blockNumber,
+    includeTransactions: true,
+  });
+
+  const fees : bigint = BigInt(1);
+
+  if (blockData) {
+    const formattedTimestamp = formatTimestamp(Number(blockData.timestamp));
+
+    return (
+      <div className="container mx-auto px-4 py-8">
+        <h1 className="text-lg font-semibold mb-4 mt-0">Block #{blockNumber.toString()}</h1>
+        <div className="shadow overflow-hidden sm:rounded-lg">
+          <div className="border-t border-gray-200 px-4 py-5 sm:p-0">
+            <dl className="sm:divide-y sm:divide-gray-200">
+            <div className="py-4 sm:py-5 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-6">
+                <dt className="text-sm font-medium ">Block Height:  </dt>
```

Comment:
Use `DescriptionListItem` from `@/components/lib/description-list-item.tsx` to ensure consistency between pages.